exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/vendors/FunctionProvider.kt

```diff
@@ -721,71 +724,13 @@ abstract class FunctionProvider {
     }
 
     /**
-     * Returns the columns to be used in the conflict condition of an upsert statement.
-     */
-    protected fun getKeyColumnsForUpsert(table: Table, vararg keys: Column<*>): List<Column<*>>? {
-        return keys.toList().ifEmpty {
-            table.primaryKey?.columns?.toList() ?: table.indices.firstOrNull { it.unique }?.columns
-        }
-    }
-
-    /** Returns the columns to be used in the update clause of an upsert statement. */
-    protected fun getUpdateColumnsForUpsert(
-        dataColumns: List<Column<*>>,
-        toExclude: List<Column<*>>?,
-        keyColumns: List<Column<*>>?
-    ): List<Column<*>> {
-        val updateColumns = toExclude?.let { dataColumns - it.toSet() } ?: dataColumns
-        return keyColumns?.let { keys ->
-            updateColumns.filter { it !in keys }.ifEmpty { updateColumns }
-        } ?: updateColumns
-    }
-
-    /**
-     * Appends the complete default SQL insert (no ignore) command to [this] QueryBuilder.
-     */
-    protected fun QueryBuilder.appendInsertToUpsertClause(table: Table, data: List<Pair<Column<*>, Any?>>, transaction: Transaction) {
-        val valuesSql = if (data.isEmpty()) {
-            ""
-        } else {
-            data.appendTo(QueryBuilder(true), prefix = "VALUES (", postfix = ")") { (column, value) ->
-                registerArgument(column, value)
-            }.toString()
-        }
-        val insertStatement = insert(false, table, data.unzip().first, valuesSql, transaction)
-
-        +insertStatement
-    }
-
-    /**
-     * Appends an SQL update command for a derived table (with or without alias identifiers) to [this] QueryBuilder.
+     * Appends to a [queryBuilder] the SQL syntax for a column that represents the same values from the INSERT clause
```

Comment:
yep, the name is not clear. Let's discuss the purpose and try to find something better

Reply:
in the parameter position it looks better